src/jsmn/jsmn.c

```diff
@@ -329,6 +329,18 @@ int jsmn_isNull(const jsmntok_t *tok)
     return strncmp("null", tok->data, 3) == 0;
 }
 
+const jsmntok_t * jsmn_find_node_type(const jsmntok_t *node, const jsmntype_t node_type)
+{
+    if (NULL == node)
+            return NULL;
+
+    for (; node->start || node->end; ++node)
```

Comment:
this just emulates the other functions in the jsmn library we incorporated.  It sweeps through the current group of nodes in the json structure from beginning to end.